src/mappings/common.ts

```diff
@@ -64,6 +64,10 @@ export function eventId(event: SubstrateEvent): string {
     return `${blockNumber(event)}-${event.idx}`
 }
 
+export function previouseEventId(event: SubstrateEvent): string {
+    return `${blockNumber(event)}-${Number(event.idx) - 1}`
```

Comment:
There is no guarantee that events will be handled in ascending order.

I catch the situation on the Karura network when it handle all currencies.Transferred events and then all tokens.Transfer from this extrinsic:
https://karura.subscan.io/event?extrinsic=2207228-2
that leads to problem when previouseEvent handle before following

https://user-images.githubusercontent.com/40560660/177055587-26adca92-429f-4918-a580-c1b331f156f7.mp4